src/python/pants/option/options_bootstrapper.py

```diff
@@ -160,13 +165,23 @@ def verify_configs_against_options(self, options):
     :param options: Fully bootstrapped valid options.
     :return: None.
     """
+    def is_valid_option(prefix, option, valid_options_under_scope):
+      if option in valid_options_under_scope:
+        return True
+      if prefix and '_'.join((prefix, option)) in valid_options_under_scope:
```

Comment:
Just passing through, but I'm confused by this bit. Is there a test for the prefix addition cases here? Can prefixes include `.`s and `-`s?

Also, this has the consequence of allowing options prefixed with their section name to elide the section name. For example, the ivy subsystem has --ivy-settings and --ivy-profile declared as options. With this change pants.ini can refer to them as either `ivy_settings:` and `ivy_profile:` or as `settings:` and `profile:`, without the prefix.

Reply:
fixed.

Reply:
Thanks!